crates/loader/src/local.rs

```diff
@@ -111,9 +111,14 @@ impl LocalLoader {
         component: v2::Component,
     ) -> Result<LockedComponent> {
         outbound_http::allowed_http_hosts::parse_allowed_http_hosts(&component.allowed_http_hosts)?;
+        if let Some(hosts) = &component.allowed_outbound_hosts {
+            spin_outbound_networking::AllowedHosts::parse(hosts)
```

Comment:
If we allow `parse` to accept an Option then we can avoid having to check at the call site

Reply:
`parse` taking `Option` seems a bit too tied to the fact that the manifest file currently is an optional field. We're discussing in the other thread about this not being the case potentially so I'd rather keep the slightly more verbose version here.